app/scripts/services/Citations.svc.js

```diff
@@ -1,14 +1,52 @@
 'use strict';
 
-angular.module('yourStlCourts').factory('Citations', function ($resource) {
+angular.module('yourStlCourts').factory('Citations', function ($resource,$q) {
   var CitationResource = $resource('citations/:id');
 
   function find(parameters) {
-    return CitationResource.get(parameters).$promise;
+    var deferred = $q.defer();
+    CitationResource.get(parameters,function(citations){
```

Comment:
So I modified this code a little and removed the reject callback.
I want to keep the datestring to date conversion in the service for a couple of reasons:
1) enforces that a date will be used in code
2) both citationInfo and paymentOptions use the Citation service.  So this way I only have to do the conversions once in the service instead of in multiple places in the code.